dask_planner/src/sql/optimizer.rs

```diff
@@ -39,7 +39,12 @@ pub struct DaskSqlOptimizer {
 impl DaskSqlOptimizer {
     /// Creates a new instance of the DaskSqlOptimizer with all the DataFusion desired
     /// optimizers as well as any custom `OptimizerRule` trait impls that might be desired.
-    pub fn new() -> Self {
+    pub fn new(
+        fact_dimension_ratio: Option<f64>,
+        max_fact_tables: Option<usize>,
+        preserve_user_order: Option<bool>,
+        filter_selectivity: Option<f64>,
```

Comment:
It's kind of annoying/bulky to have to pass in these parameters every time we do a `DaskSqlOptimizer::new()`, but since there's really only 2 places in our code where we do this, I figured it's not too bad.